controllers/volumereplicationgroup_controller.go

```diff
@@ -256,8 +381,49 @@ func (v *VolumeReplicationGroupReconciler) HandlePersistentVolumeClaims(
 	return pvcList, nil
 }
 
+// handlePersistentVolumeClaims creates VolumeReplication CR for each pvc
+// from pvcList. If it fails (even for one pvc), then requeue is set to true.
+// For now, keeping creation of VolumeReplication CR and backing up of PV
+// metadata in separate functions and calling them separately. In future,
+// if creation of VolumeReplication CR and backing up of pv metadata should
+// happen together (i.e for each pvc create VolumeReplication CR and then
+// backup corresponding PV metadata), then it can be put in a single function.
+func (v *VolumeReplicationGroupReconciler) handlePersistentVolumeClaims(
+	ctx context.Context,
+	volRepGroup *ramendrv1alpha1.VolumeReplicationGroup,
+	pvcList *corev1.PersistentVolumeClaimList) bool {
+	if requeue := v.createVolumeReplicationResources(ctx, volRepGroup, pvcList); requeue {
+		log.Error("failed to get or create VolumeReplication CRs for pvc list")
+
+		return requeue
+	}
+
+	if requeue := v.handlePersistentVolumes(ctx, pvcList); requeue {
+		log.Error("failed to handle Persistent Volumes for pvc list")
+
+		return requeue
+	}
+
+	return false
```

Comment:
(nit) I prefer to use `return !requeue`.  It reads better when the previously you had `return requeue` and then the last one is `return false`.  So you may need to declar requeue outside the `if`

Reply:
From your suggestion it seems you are telling me to declare requeue (initialized to true?) in the beginning and then at the
end "return !requeue".

That might create some issues. Assume requeue is initialized to true at the time of declaration. Then both of the if statements are false. So ideally we should be returning false from the function. But doing "return !requeue" leads to returning true.

What I can do is, initialize requeue to false at the time of declaration and at the end do just "return requeue". That way, if any of the if conditions are true, then immediately, requeue is set to true and that is returned then and there itself. By the time we reach end of the function, both the if conditions are false, hence requeue would still be false and safely we can return it.

NOTE: I am not sure about scope of the values returned in if blocks for the variables that are declared outside the if blocks  
            in Go. 

Let me know your thoughts.

Reply:
The reason I suggested the change is because these requeues are coming from everywhere and it becomes easily messy, especially when the linter forces you to split a function.  What I suggested was very simple.  Here it is:
```
func (v *VolumeReplicationGroupReconciler) handlePersistentVolumeClaims(
	ctx context.Context,
	volRepGroup *ramendrv1alpha1.VolumeReplicationGroup,
	pvcList *corev1.PersistentVolumeClaimList) bool {

        const requeue = true

	if requeueAction := v.createVolumeReplicationResources(ctx, volRepGroup, pvcList); requeueAction {
		log.Error("failed to get or create VolumeReplication CRs for pvc list")

		return requeue
	}

	if requeueAction := v.handlePersistentVolumes(ctx, pvcList); requeueAction {
		log.Error("failed to handle Persistent Volumes for pvc list")

		return requeue
	}

	return !requeue
}
```
That's all.  I named the tmp variable requeueAction just to show what I mean.  You can name that anything meaningful and even `rq`
But again, it was a small suggestion only and it will not hold up the PR.

Reply:
ACK.